Source/Models/BrickDimension.swift

```diff
@@ -18,39 +18,106 @@ public struct BrickSize {
     }
 }
 
+public typealias RangeDimensionPair = (dimension: BrickDimension, minimumLength: CGFloat)
+
+public struct BrickRangeDimension {
+    
+    internal var dimensionPairs : [RangeDimensionPair]
+    
+    public init(minimum dimension: BrickDimension, additionalRangePairs: [RangeDimensionPair]?) {
+        dimensionPairs = [(dimension, 0)]
+        if let additionalRangePairs = additionalRangePairs {
+            dimensionPairs.append(contentsOf: additionalRangePairs.sorted {
+                $0.minimumLength < $1.minimumLength
+            })
+        }
+    }
+    
+    public func dimension(forWidth width: CGFloat?) -> BrickDimension {
+        guard let width = width  else {
+            return dimensionPairs.first!.dimension
+        }
+        var dimension = dimensionPairs.first?.dimension
+        for RangeDimensionPair in dimensionPairs {
+            if RangeDimensionPair.minimumLength > width {
+                break
+            }
+            else {
+                dimension = RangeDimensionPair.dimension
+            }
+        }
+        return dimension!
+    }
+}
+
+internal func almostEqualRelative(first: CGFloat, second: CGFloat, maxRelDiff: CGFloat = CGFloat.ulpOfOne) -> Bool
+{
+    // Calculate the difference.
+    let diff = fabs(first - second);
+    let a = fabs(first)
+    let b = fabs(second)
+    // Find the largest
+    let largest = (b > a) ? b : a
+    
+    if diff <= largest * maxRelDiff {
+        return true
+    }
+    return false
+}
+
+public func ==(lhs: RangeDimensionPair, rhs: RangeDimensionPair) -> Bool {
+    return lhs.dimension == rhs.dimension && almostEqualRelative(first: lhs.minimumLength, second: rhs.minimumLength)
+}
+
+public func ==(lhs: BrickRangeDimension, rhs: BrickRangeDimension) -> Bool {
```

Comment:
There were but the last commit ended up modifying the coverage. I'll add additional tests to handle the additional cases.